### PR TITLE
Add tests for profile.clj

### PR DESCRIPTION
**Pull request by kelvinqian00**
Add these tests for profile.clj:
- Manual tests for `profiles->map` and `pattern-zip`
- Generative tests for `walk-once` (which covers `rand-pattern-zip`)
- Slightly modified the cmi5 Profile to be able to exercise oneOrMore Patterns

### CODE REVIEW

File: dev-resources/profiles/cmi5/fixed.json [hunk 1]
```diff
@@ -705,7 +705,7 @@
                 "en": "This pattern provides an option to use zero or more of the alernates for the Typical Session pattern."
             },
             "inScheme": "https://w3id.org/xapi/cmi5/v1.0",
-            "zeroOrMore": "https://w3id.org/xapi/cmi5#typicalsession"
+            "oneOrMore": "https://w3id.org/xapi/cmi5#typicalsession"
```
Comment (milt) [inline dev-resources/profiles/cmi5/fixed.json:708]: does this track some change in the upstream CMI profile? just curious

File: src/test/com/yetanalytics/datasim/xapi/profile_test.clj [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+         :failed-session
+         :cmi5/failed-session
+         :no-result-session
+         :cmi5/no-result-session
+         :passed-session
+         :cmi5/passed-session
+         :completion-no-success-session
+         :cmi5/completion-no-success-session
+         :waived-session
+         :cmi5/waived-session))
```
Comment (milt) [inline src/test/com/yetanalytics/datasim/xapi/profile_test.clj:150]: You may want to try using regular `def` and symbol names for your test-specific specs in this namespace.
At least in spec 1 (dunno about 2), spec registry by keyword is _global_ and thus we try to keep it out of tests. The good news is you could just as easily write the spec on line 53 as:

    (def cmi5-launched (partial is-cmi5-id? "launched"))

and just use symbols down the line. You only actually need to `s/def` keywords for specs that will be _keys in map specs_

**Review by milt — COMMENTED** [review]
The tests themselves are really good (and much-needed, thanks!) Just see the note about using keyword specs vs just `def`ing specs in the ns

File: dev-resources/profiles/cmi5/fixed.json (hunk 1 above)
Comment (kelvinqian00) [reply dev-resources/profiles/cmi5/fixed.json:708]: No, this change is 100% local. So this makes it _not_ the actual cmi5 profile, but instead a variant with a oneOrMore Pattern. None of the Profiles in `dev-resources` originally had a oneOrMore Pattern, hence why I had to make this change.

File: src/test/com/yetanalytics/datasim/xapi/profile_test.clj (hunk 2 above)
Comment (kelvinqian00) [reply src/test/com/yetanalytics/datasim/xapi/profile_test.clj:150]: Ok I can do that

File: dev-resources/profiles/cmi5/fixed.json (hunk 1 above)
Comment (kelvinqian00) [reply dev-resources/profiles/cmi5/fixed.json:708]: The alternative would've been to make a custom Profile with all five different Patterns, but that seemed overkill.

File: src/test/com/yetanalytics/datasim/xapi/profile_test.clj (hunk 2 above)
Comment (kelvinqian00) [reply src/test/com/yetanalytics/datasim/xapi/profile_test.clj]: Done

**Pull request merged**